src/superface-test.ts

```diff
@@ -331,7 +376,41 @@ export class SuperfaceTest {
         );
       }
 
-      await writeRecordings(this.recordingPath, definitions);
+      const recordingExists = await exists(this.composeRecordingPath());
+
+      if (!recordingExists) {
+        // recording file does not exist -> record new traffic
+        await writeRecordings(this.composeRecordingPath(), definitions);
+      } else {
+        // recording file exist -> record and compare new traffic
+        const oldRecording = await readFileQuiet(this.composeRecordingPath());
+
+        if (oldRecording === undefined) {
+          throw new UnexpectedError('Reading recording file failed');
+        }
+
+        const oldRecordingDefs = JSON.parse(
+          oldRecording
+        ) as RecordingDefinitions;
+
+        // Match new HTTP traffic to saved for breaking changes
+        const matching = await matchTraffic(oldRecordingDefs, definitions);
+
+        if (matching) {
+          // do not save new recording as there were no breaking changes found
+        } else {
+          // save new recording next to old one as there were breaking changes and map need to be updated
+          // Alert changes
+          // can consist of call to some monitoring service or just calling some custom webhook
+
+          // Save new recording as unsupported
+          await writeRecordings(
```

Comment:
It took me all the way down here to understand why the recording is called `unsupported` ☺️